app/models/manageiq/providers/ansible_tower/shared/automation_manager/configuration_script_source.rb

```diff
@@ -33,9 +33,13 @@ def refresh_in_provider
       log_header = "updating project #{project.id} (#{self.class.name} #{id})"
       _log.info "#{log_header}..."
       Timeout.timeout(5.minutes) do
-        while project_update.finished.blank?
-          sleep REFRESH_ON_TOWER_SLEEP
+        loop do
           project_update = project_update.api.project_updates.find(project_update.id)
+          # the sleep here is also needed because tower needs some time to actually propagate it's updates
+          # if we would return immediately it _could_ be that the we get the old playbooks
+          # the whole sleep business is a workaround anyway until we get proper polling via the PR mentioned above
+          sleep REFRESH_ON_TOWER_SLEEP
+          break if project_update.finished.present?
```

Comment:
my guess is, the update has finished (update is a job in tower), but it takes some ms to propagate those changes to the playbook tables...